src/syntax/ppx_sedlex.ml

```diff
@@ -211,22 +211,26 @@ let gen_state lexbuf auto i (trans, final) =
   let body () =
     pexp_match ~loc
       (appfun (partition_name partition)
-         [[%expr Sedlexing.__private__next_int [%e evar ~loc lexbuf]]])
+         [[%expr Sedlexing.__private__next_int [%e lexbuf]]])
       (cases
       @ [
           case
             ~lhs:[%pat? _]
             ~guard:None
-            ~rhs:[%expr Sedlexing.backtrack [%e evar ~loc lexbuf]];
+            ~rhs:[%expr Sedlexing.backtrack [%e lexbuf]];
         ])
   in
   let ret body =
+    let lhs =
+      match lexbuf with
+        | { pexp_desc = Pexp_ident { txt = Lident ident; _ }; pexp_loc } ->
+            pvar ~loc:pexp_loc ident
+        | _ -> assert false
```

Comment:
You could use the `Longident.t loc` for the location (and get it from `method! expression` as before).
That way you avoid the `assert false`

Reply:
I still need the string inside the simple `ident` to construct the pattern

Reply:
What about https://github.com/hhugo/sedlex/compare/fixloc...smuenzel:sedlex:fixloc-explore ?
Shorter and we encode our invariants in the type system, rather than in `assert false`

Reply:
looks good, I've cherry-pick you commit